frame/timestamp/src/lib.rs

```diff
@@ -167,9 +174,16 @@ decl_module! {
 			<T::OnTimestampSet as OnTimestampSet<_>>::on_timestamp_set(now);
 		}
 
+		/// dummy `on_initialize` to return the weight used in `on_finalize`.
+		fn on_initialize() -> Weight {
+			// weight of `on_finalize`
+			6_000_000
```

Comment:
why this is not `T::DbWeight::write(1)`?

Reply:
This takes `DidUpdate`, which is empty at the initialization of the block, populated as the first extrinsic in the block, and then removed again at the finalization of the block.

No value actually gets written to the DB layer. Everything happens within one block in-memory.